core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/ActivityIntegrationPoint.kt

```diff
@@ -44,4 +48,44 @@ open class ActivityIntegrationPoint(
             activity.finish()
         }
     }
+
+    companion object {
+        private val integrationPoints = WeakHashMap<AppCompatActivity, IntegrationPoint>()
+
+        fun createIntegrationPoint(
+            activity: AppCompatActivity,
+            savedInstanceState: Bundle?,
+        ): ActivityIntegrationPoint =
+            ActivityIntegrationPoint(activity, savedInstanceState)
```

Comment:
shouldn't we check for existing integration point for this activity before creating it?

Reply:
The existing API doesn't work this way. We could do a getOrCreate or something similar, but I wanted to remain reasonably consistent.